metals/src/main/scala/scala/meta/internal/metals/ScalafixProvider.scala

```diff
@@ -0,0 +1,147 @@
+package scala.meta.internal.metals
+
+import java.nio.file.FileSystems
+import java.nio.file.Path
+import java.util
+import java.util.Collections
+
+import scala.collection.mutable
+import scala.util.Try
+
+import scala.meta._
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.io.AbsolutePath
+
+import org.eclipse.lsp4j.MessageType
+import org.eclipse.{lsp4j => l}
+import scalafix.interfaces.Scalafix
+import scalafix.interfaces.ScalafixArguments
+
+case class ScalafixProvider(
+    buildTargets: BuildTargets,
+    buffers: Buffers,
+    scalafixConfigPath: RelativePath,
+    workspace: AbsolutePath,
+    embedded: Embedded,
+    statusBar: StatusBar,
+    icons: Icons,
+    languageClient: MetalsLanguageClient
+) {
+  import ScalafixProvider._
+  private val scalafixCache = mutable.Map.empty[String, Scalafix]
+
+  def organizeImports(file: AbsolutePath): List[l.TextEdit] = {
+    val fileInput = file.toInput
+    val unsavedFile = file.toInputFromBuffers(buffers)
+    if (fileInput.text.diff(unsavedFile.text).nonEmpty) {
```

Comment:
Why use `diff` instead of just equals or calculating md5 from something like:
```scala
def md5(text : String) = {
  val digest = MessageDigest.getInstance("MD5")
  digest.update(text)
  MD5.bytesToHex(digest.digest())
}
```

Reply:
Weirdly the equality fails on windows.

Reply:
And then for computing md5, you still need to read the file (to get the String). So I thought from a performance point of view, it's the same? 
I will implement this with md5.

Reply:
Not 100% sure which one is faster, but md5 is used already to check if a file changed and diff seems a bit more involved.

Reply:
so md5 is not working on windows ...

Reply:
That's super weird :/ I will try out later at home on windows.

Reply:
Thanks